gf2_192/src/gf2_192poly.rs

```diff
@@ -0,0 +1,243 @@
+        t = t + value_at_zero;
+        s = Gf2_192::invert(s);
+        t = t * s;
+        result.add_monic_times_constant(vanishing_poly, t);
+
+        Ok(result)
+    }
+
+    /// Evaluates polynomial at the given point `x`, where `x` represents the last byte of a field
+    /// element (all other bits are assumed to be 0)
+    pub fn evaluate(&self, x: i8) -> Gf2_192 {
+        let mut res = self.coefficients[self.degree];
+        if self.degree > 0 {
+            for d in (0..=(self.degree - 1)).rev() {
+                res = Gf2_192::mul_by_i8(res, x);
+                res = res + self.coefficients[d];
+            }
+        }
+        res
+    }
+
+    /// Returns Vec consisting of the concatenation of all the coefficients of the polynomial.
+    ///  - Degree-zero coefficient is located at index 0
+    ///  - Each coefficient takes 24 bytes for a total of `(self.degree+1)*24` bytes
+    pub fn to_i8_vec(&self) -> Vec<i8> {
+        let mut res: Vec<_> = std::iter::repeat(0).take((self.degree + 1) * 24).collect();
+        for i in 0..self.degree {
+            #[allow(clippy::unwrap_used)]
+            self.coefficients[i].to_i8_slice(&mut res, i * 24).unwrap();
+        }
+        res
+    }
+
+    /// Adds r*p to `self`. Assumes:
+    ///  - p is monic
+    ///  - self.coefficients.len() > p.degree
+    ///  - p.degree == self.degree + 1 or (self == 0 and p == 1)
+    fn add_monic_times_constant(&mut self, p: Gf2_192Poly, r: Gf2_192) {
+        let mut _t = Gf2_192::new();
+        for i in 0..p.degree {
+            _t = p.coefficients[i] * r;
+            self.coefficients[i] = self.coefficients[i] + _t;
+        }
+        self.degree = p.degree;
+        self.coefficients[self.degree] = r;
+    }
+
+    /// Multiply `self` by `x + r`. Assumes that `self` is monic (i.e.
+    /// `self.coefficients[self.degree] == 1`)
+    fn multiply_by_linear_binomial(&mut self, r: i8) {
+        self.degree += 1;
+        self.coefficients[self.degree] = Gf2_192::from(1);
+        for i in (1..self.degree).rev() {
+            self.coefficients[i] = Gf2_192::mul_by_i8(self.coefficients[i], r);
+            self.coefficients[i] = self.coefficients[i] + self.coefficients[i - 1];
+        }
+        self.coefficients[0] = Gf2_192::mul_by_i8(self.coefficients[0], r);
+    }
+
+    /// Constructs a constant polynomial (degree 0) which takes value of `constant_term`.
+    /// `max_degree` specifies the maximum degree of the polynomial (to allocate space).
+    fn make_constant(max_degree: usize, constant_term: i32) -> Gf2_192Poly {
+        let mut coefficients: Vec<_> = std::iter::repeat_with(Gf2_192::new)
+            .take(max_degree + 1)
+            .collect();
+        coefficients[0] = Gf2_192::from(constant_term);
+        let degree = 0;
+        Gf2_192Poly {
+            degree,
+            coefficients,
+        }
+    }
+}
+
+impl<'a> TryFrom<CoefficientsByteRepr<'a>> for Gf2_192Poly {
```

Comment:
Could you pull in here the explanation for this from 
https://github.com/ScorexFoundation/sigmastate-interpreter/blob/262a14d17ae3090fa97d1de1effd0b338562e4db/sigmastate/src/main/java/gf2t/GF2_192_Poly.java#L40-L48
and/or add it to the `CoefficientByteRepr` type.

Reply:
Sure.